src/Logic.hs

```diff
@@ -561,19 +570,29 @@ pushCandidate (pullRequestId, pullRequest) newHead state =
       Username approvedBy = approver approval
       commentToUser = leaveComment pullRequestId . (<>) ("@" <> approvedBy <> " ")
   in assert (isJust $ Pr.approval pullRequest) $ do
-    pushResult <- if Pr.needsTag $ Pr.approvedFor approval
+    let approvalKind = Pr.approvedFor approval
+    pushResult <- if Pr.needsTag $ approvalKind
       then do
         versionOrBadTag <- getLatestVersion newHead
         case versionOrBadTag of
           Left (TagName bad) -> do
             commentToUser ("Sorry, I could not tag your PR. The previous tag `" <> bad <> "` seems invalid")
             tryPromote prBranch newHead
           Right v -> do
-            (tagResult, pushResult) <- tryPromoteWithTag prBranch newHead (versionToTag $ v + 1)
+            let previousTag = versionToTag v
+            mChangelog <- getChangelog previousTag newHead
+            let
+              tagName = versionToTag $ v + 1
+              changelog = maybe "no changelog" id mChangelog
```

Comment:
Suggestion: AIUI we should only get a `Nothing` here if Hoff fails to get the changelog, not if there is no changelog, in the unlikely case where the previous tag already points to the commit that Hoff tagged I think it should give an empty string as changelog with exit code 0 (see e.g. `git shortlog HEAD..HEAD`).

I suggest making it clearer that something failed:

```suggestion
              changelog = maybe "Failed to get changelog" id mChangelog
```